### PR TITLE
Add PyPI release workflow

### LINKED ISSUE #6: Add pypi release workflow

### PR DESCRIPTION
**Pull request by dcherian**
Closes #6

### CODE REVIEW

File: .github/workflows/pypi-release.yaml [hunk 1]
```diff
@@ -0,0 +1,97 @@
+name: Build and Upload to PyPI
+on:
+  release:
+    types:
+      - published
+  push:
+    tags:
+      - 'v*'
+
```
Comment (andersy005) [inline .github/workflows/pypi-release.yaml:9]: we could also add the following to catch issues before releases 

```suggestion
    branches:
      - master

  pull_request:
    branches:
      - master

concurrency:
  group: ${{ github.workflow }}-${{ github.ref }}
  cancel-in-progress: true
```

File: .github/workflows/pypi-release.yaml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+          if [ -f dist/cupy-xarray-0.0.0.tar.gz ]; then
+            echo "❌ INVALID VERSION NUMBER"
+            exit 1
+          else
+            echo "✅ Looks good"
+          fi
+      - uses: actions/upload-artifact@v3
+        with:
+          name: releases
+          path: dis
```
Comment (andersy005) [inline .github/workflows/pypi-release.yaml:47]:
```suggestion
          path: dist
```

File: .github/workflows/pypi-release.yaml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+      - name: List contents of built dist
+        run: |
+          ls -ltrh
+          ls -ltrh dist
+
+      - name: Verify the built dist/wheel is valid
+        if: github.event_name == 'push'
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install dist/xarray*.whl
```
Comment (andersy005) [inline .github/workflows/pypi-release.yaml:70]:
```suggestion
          python -m pip install dist/cupy-xarray*.whl
```

File: .github/workflows/pypi-release.yaml [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+          python -m cupy_xarray.__version__
+
+      - name: Publish package to TestPyPI
+        if: github.event_name == 'push'
+        uses: pypa/gh-action-pypi-publish@v1.5.1
+        with:
+          user: __token__
+          password: ${{ secrets.TESTPYPI_TOKEN }}
+          repository_url: https://test.pypi.org/legacy/
+          verbose: true
```
Comment (andersy005) [inline .github/workflows/pypi-release.yaml:80]: Do we need to push to TestPyPI? If so, happy to create the necessary token.

File: .github/workflows/pypi-release.yaml [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    steps:
+      - uses: actions/download-artifact@v3
+        with:
+          name: releases
+          path: dist
+      - name: Publish package to PyPI
+        uses: pypa/gh-action-pypi-publish@v1.5.1
+        with:
+          user: __token__
+          password: ${{ secrets.PYPI_TOKEN }}
```
Comment (andersy005) [inline .github/workflows/pypi-release.yaml:96]: I added the necessary secret to the repository

File: .github/workflows/pypi-release.yaml (hunk 4 above)
Comment (dcherian) [reply .github/workflows/pypi-release.yaml:80]: eh just copied over the xarray workflow

Comment (dcherian) [reply .github/workflows/pypi-release.yaml]: OK deleted the test bit. We can't test without a GPU enabled machine

**Pull request merged**